From 210d31f4244151f91a0845d02fcacfb949e36f1c Mon Sep 17 00:00:00 2001
From: Akshay Gupta <Akshay.Gupta@amd.com>
Date: Thu, 19 Aug 2021 16:29:17 +0530
Subject: [PATCH] ARM:dts:aspeed: Add dts support for sbrmi module for
 Onyx/Quartz

- APML over I2C, node for sbrmi in Onyx & Quartz
- node for sbtsi & sbrmi in Ruby

Signed-off-by: Akshay Gupta <Akshay.Gupta@amd.com>
---
 arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts   |  5 ++++
 arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts | 10 +++++++
 arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts   | 30 +++++++++++++++++++++
 3 files changed, 45 insertions(+)

diff --git a/arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts b/arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts
index 0b47dfa83a7b..4fe6bc2f80e9 100644
--- a/arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts
+++ b/arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts
@@ -438,6 +438,11 @@
 		compatible = "amd,sbtsi";
 		reg = <0x4c>;
 	};
+	// sbrmi
+	sbrmi@3c {
+		compatible = "amd,sbrmi";
+		reg = <0x3c>;
+	};
 };

 &espi_ctrl {
diff --git a/arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts b/arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts
index e31e0ee9404c..e56ec2046f8b 100644
--- a/arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts
+++ b/arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts
@@ -464,6 +464,11 @@
 		compatible = "amd,sbtsi";
 		reg = <0x4c>;
 	};
+	// sbrmi
+	sbrmi@3c {
+		compatible = "amd,sbrmi";
+		reg = <0x3c>;
+	};
 };

 &i2c3 {
@@ -474,6 +479,11 @@
 		compatible = "amd,sbtsi";
 		reg = <0x48>;
 	};
+	// sbrmi
+	sbrmi@38 {
+		compatible = "amd,sbrmi";
+		reg = <0x38>;
+	};
 };

 &espi_ctrl {
diff --git a/arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts b/arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts
index f38c241fe97f..a2f2a89b561e 100644
--- a/arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts
+++ b/arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts
@@ -321,6 +321,36 @@
 	status = "okay";
 };

+&i2c2 {
+	// P0 APML
+	status = "okay";
+
+	sbtsi@4c {
+		compatible = "amd,sbtsi";
+		reg = <0x4c>;
+	};
+	// sbrmi
+	sbrmi@3c {
+		compatible = "amd,sbrmi";
+		reg = <0x3c>;
+	};
+};
+
+&i2c3 {
+	// P1 APML
+	status = "okay";
+
+	sbtsi@48 {
+		compatible = "amd,sbtsi";
+		reg = <0x48>;
+	};
+	// sbrmi
+	sbrmi@38 {
+		compatible = "amd,sbrmi";
+		reg = <0x38>;
+	};
+};
+
 &espi_ctrl {
 	status = "okay";

--
2.17.1
